frontend/src/pages/associations/elections/Edit.tsx

```diff
@@ -0,0 +1,416 @@
+import React, {useContext, useState} from "react";
+import {Form, Button, Col, InputGroup, Alert} from "react-bootstrap";
+import { api, useBetterQuery } from "../../../services/apiService";
+import { useParams } from "react-router";
+import { ToastContext, ToastLevel } from "../../../utils/Toast";
+import {Choice, Election} from "../../../models/associations/election";
+import {PageTitle} from "../../../utils/common";
+import {User} from "../../../models/user";
+
+
+export const AssociationCreateElection = ({ association, ...props }) => {
+    const election: Election = {
+        name: "",
+        choices: [{
+            id: 0, election: 0, name:""
+        }, {
+            id:0, election: 0, name:""
+        }],
+        registeredVoters: ['17bocquet', '17wan-fat'], //to-do : change to []
+        startsAt: new Date(),
+        endsAt: new Date(),
+        maxChoicesPerBallot: 1,
+        association: association.id,
+    };
+
+    return (
+        <>
+            <PageTitle className={"mt-6"}>Création d'une élection</PageTitle>
+            <EditElection {...props} election={election} association={association} edit={false}/>
+        </>
+    )
+};
+
+export const AssociationEditElection = ({ association, ...props }) => {
+    const { electionId } = useParams<{electionId: string}>();
+
+    const { data: election, status, error } = useBetterQuery<Election>(
+        "election.get",
+        api.elections.get,
+        electionId
+    );
+
+    if (status === "loading") return "Chargement en cours...";
+    else if (status === "error") return `Une erreur est apparue: ${error}`;
+    else if (election)
+        return (
+            <>
+                <PageTitle className={"mt-6"}>Modification d'une élection</PageTitle>
+                <p>Attention ! C'est la supprimer et la remplacer. Tous les votes seront perdus. </p>
+                <EditElection {...props} election={election} association={association} edit={true}/>
+            </>
+        );
+    return null;
+};
+
+const EditElection = ({ election, association, ...props }) => {
```

Comment:
I am sorry, I think I haven't told you: we use [Formik](https://jaredpalmer.com/formik/docs/overview) for our forms. Another library, but it is really useful, especially for handling errors.

Some work is needed to make it work cleanly with tabler.io, and I have created some components in my branch `pollsFrontend`. We can talk about how to (re)use them.